MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm

```diff
@@ -959,9 +1007,12 @@ static MTLVertexFormat mvkAdjustFormatVectorToSize(MTLVertexFormat format, uint3
 		auto& funcRslts = func.shaderConversionResults;
 		_needsVertexSwizzleBuffer = funcRslts.needsSwizzleBuffer;
 		_needsVertexBufferSizeBuffer = funcRslts.needsBufferSizeBuffer;
+		_needsVertexDynamicOffsetBuffer = funcRslts.needsDynamicOffsetBuffer;
 		_needsVertexOutputBuffer = funcRslts.needsOutputBuffer;
 	}
 
+	addMTLArgumentEncoders(func, pCreateInfo, shaderContext, kMVKShaderStageVertex);
```

Comment:
Yes. I was treating the extraction of the arg encoder from the `MTLFunction` as independent of the index width...similar to how the several `_needsVertex...` flags are extracted from `shaderConversionResults`.

From my review of `spirv_msl.cpp`, I don't see that the index width will have an effect on the arg encoder content. But...you definitely have a much better handle than I do on how the three index width variations are compiled. Are you thinking the index width might vary the arg encoder?

It's not a critical item for now, as this operation is actually not used because we don't properly support arg encoders pulled from the `MTLFunction` yet. We can leave this for further testing as we expand the platform support for Metal argument buffers.

Reply:
Yeah, I think you're right, and it shouldn't depend on the index type.